### ISSUE TITLE
After the last issues in a file reported by ansible-lint are fixed, those still get shown

### ISSUE THREAD

**Issue by flodabo** [ansible-community/vscode-ansible#95]
**Describe the bug**

After the last issue(s) within a .yml file reported by ansible-lint are fixed and the file saved, the issue(s) still gets shown inside the editor as well as the problems list, until the the file gets closed.

**To Reproduce**

1. Open a .yml file with one or more issues reported by ansible-lint
2. Fix all the issues and save the file
3. See the issues fixed in step 2 still present in the problems list of VSCode as well as in the editor
4. Close the file. Watch issues disappearing from the problem list

***Example playbook for testing***

```yaml
- hosts: localhost
  vars: 
    testvar: "hello"
  tasks:
    - name: "test one {{testvar}}"
      debug:
        msg: "this is a test"
```

**Expected behavior**

The issues should not be shown any longer after the issues get fixed and the file gets saved, without having to close the file.

**Platform:**

tested on Fedora Linux and Windows 10

***test system 1: Windows***

```
Version: 1.56.0 (system setup)
Commit: cfa2e218100323074ac1948c885448fdf4de2a7f
Date: 2021-05-04T22:09:06.405Z
Electron: 12.0.4
Chrome: 89.0.4389.114
Node.js: 14.16.0
V8: 8.9.255.24-electron.0
OS: Windows_NT x64 10.0.19042
```

remote ssh is used to connect to a Debian 10 system
ansible-lint 5.0.8 using ansible 2.10.9 on the remote system

***test system 2: Fedora***

```
Version: 1.56.0
Commit: cfa2e218100323074ac1948c885448fdf4de2a7f
Date: 2021-05-04T21:58:14.757Z
Electron: 12.0.4
Chrome: 89.0.4389.114
Node.js: 14.16.0
V8: 8.9.255.24-electron.0
OS: Linux x64 5.11.18-300.fc34.x86_64
```

working on local files
ansible-lint 5.0.8 using ansible 2.10.9

Both test systems show the same behavior.

**Additional context**

When fixing the issues one by one and saving in between, the fixed issues get removed from the problem list as excepted. This only happens if there are no further issues found in the file.

**Comment by axymen** [comment]
I have the same issue. VSCodium on Ubuntu 21.04:
Version: 1.56.0
Commit: cfa2e218100323074ac1948c885448fdf4de2a7f
Date: 2021-05-09T15:54:41.116Z
Electron: 12.0.4
Chrome: 89.0.4389.114
Node.js: 14.16.0
V8: 8.9.255.24-electron.0
OS: Linux x64 5.11.0-16-generic snap

ansible-lint 5.0.8 using ansible 2.10.8
Ansible Language extension v0.4.2

**Comment by denisbee** [comment]
Same issue

`Ansible Language extension v0.4.3`
`ansible-lint 5.0.8 using ansible 2.10.9`
```
Version: 1.56.2
Commit: 054a9295330880ed74ceaedda236253b4f39a335
Date: 2021-05-12T17:44:30.902Z (4 days ago)
Electron: 12.0.4
Chrome: 89.0.4389.114
Node.js: 14.16.0
V8: 8.9.255.24-electron.0
OS: Darwin x64 20.4.0
```

**Comment by ssbarnea** [comment]
I confirm the bug and I would be very happy to review a PR that address it.

**Comment by dual-oo** [comment]
I've installed the 0.4.4 update and still get this issue. I used [code-server](https://github.com/cdr/code-server) with autosave on delay.
```
"files.autoSave": "afterDelay"
"files.autoSaveDelay": 1000
```

**Comment by ssbarnea** [comment]
You did not mention if manually saving the file works (expect up to 3s delay). For me it does seem to work. You may want to check the loggers and see what happens.

**Comment by dual-oo** [comment]
Thanks for reply. 
I've tested it with `"files.autoSaveDelay": 6000` and also with `"files.autoSave": "off"`. Still get an issue reported till I close/open the yaml file.

**Comment by dual-oo** [comment]
[https://github.com/ansible-community/vscode-ansible/commit/da46cd614f16e89a63a4938f12c2e21dc03b4c01](fix: clean old violations on reruns (#122) fixed the issue. Thank you!

**Comment by flodabo** [comment]
i can confirm that insattlling version 0.4.5 fixed the issue. thank you!
